iothub/device/src/Authentication/AuthenticationWithTokenRefresh.cs

```diff
@@ -24,20 +24,40 @@ public abstract class AuthenticationWithTokenRefresh : IAuthenticationMethod
         /// <summary>
         /// Creates an instance of this class.
         /// </summary>
+        /// <param name="deviceId">Device Identifier.</param>
+        /// <param name="moduleId">Module Identifier.</param>
         /// <param name="suggestedTimeToLive">
         /// The suggested time to live value for the generated SAS tokens.
         /// The default value is 1 hour.
         /// </param>
         /// <param name="timeBufferPercentage">
         /// The time buffer before expiry when the token should be renewed, expressed as a percentage of the time to live.
         /// The default behavior is that the token will be renewed when it has <see cref="DefaultSasRenewalBufferPercentage"/> percent or less of its lifespan left.
-        ///</param>
-        /// <exception cref="ArgumentOutOfRangeException">Thrown if <paramref name="suggestedTimeToLive"/> is a negative timespan, or if
-        /// <paramref name="timeBufferPercentage"/> is outside the range 0-100.</exception>
+        /// </param>
+        /// <exception cref="ArgumentNullException">Thrown when <paramref name="deviceId"/> is null.</exception>
+        /// <exception cref="ArgumentException">
+        /// Thrown when <paramref name="deviceId"/> or <paramref name="moduleId"/> is empty or whitespace.
+        /// </exception>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// Thrown if <paramref name="suggestedTimeToLive"/> is a negative timespan, or if <paramref name="timeBufferPercentage"/> is outside the range 0-100.
+        /// </exception>
```

Comment:
This is an extremely small nit (my 2c) - I liked the name "ClientAuthenticationWithTokenRefresh"

Reply:
Especially if otherwise we have a naming conflict with similar types in the service SDK.